src/cli/pm_version_command.zig

```diff
@@ -0,0 +1,605 @@
+                        "postversion",
+                        package_json_dir,
+                        pm.env,
+                        &.{},
+                        pm.options.log_level == .silent,
+                        ctx.debug.use_system_shell,
+                    );
+                }
+            }
+        }
+
+        Output.println("v{s}", .{new_version_str});
+        Output.flush();
+    }
+
+    fn findPackageDir(allocator: std.mem.Allocator, start_dir: []const u8) ![]const u8 {
+        var path_buf: bun.PathBuffer = undefined;
+        var current_dir = start_dir;
+
+        while (true) {
+            const package_json_path_z = bun.path.joinAbsStringBufZ(current_dir, &path_buf, &.{"package.json"}, .auto);
+            if (bun.FD.cwd().existsAt(package_json_path_z)) {
+                return try allocator.dupe(u8, current_dir);
+            }
+
+            const parent = bun.path.dirname(current_dir, .auto);
+            if (strings.eql(parent, current_dir)) {
+                break;
+            }
+            current_dir = parent;
+        }
+
+        return try allocator.dupe(u8, start_dir);
+    }
+
+    fn verifyGit(cwd: []const u8, pm: *PackageManager) !void {
+        if (!pm.options.git_tag_version) return;
+
+        var path_buf: bun.PathBuffer = undefined;
+        const git_dir_path = bun.path.joinAbsStringBuf(cwd, &path_buf, &.{".git"}, .auto);
+        if (!bun.FD.cwd().directoryExistsAt(git_dir_path).isTrue()) {
+            pm.options.git_tag_version = false;
+            return;
+        }
+
+        if (!isGitClean(cwd) and !pm.options.force) {
+            Output.errGeneric("Git working directory not clean.", .{});
+            Global.exit(1);
+        }
+    }
+
+    fn parseVersionArgument(arg: []const u8) struct { VersionType, ?[]const u8 } {
+        if (VersionType.fromString(arg)) |vtype| {
+            return .{ vtype, null };
+        }
+
+        const version = Semver.Version.parse(Semver.SlicedString.init(arg, arg));
+        if (version.valid) {
+            return .{ .specific, arg };
+        }
+
+        Output.errGeneric("Invalid version argument: \"{s}\"", .{arg});
+        Output.note("Valid options: patch, minor, major, prepatch, preminor, premajor, prerelease, from-git, or a specific semver version", .{});
+        Global.exit(1);
+    }
+
+    fn showHelp(ctx: Command.Context, pm: *PackageManager, cwd: []const u8) !void {
+        var path_buf: bun.PathBuffer = undefined;
+        const package_json_path = bun.path.joinAbsStringBuf(cwd, &path_buf, &.{"package.json"}, .auto);
+        const package_json_contents = bun.sys.File.readFrom(bun.FD.cwd(), package_json_path, ctx.allocator).unwrap() catch |err| {
+            Output.errGeneric("Failed to read package.json: {s}", .{@errorName(err)});
+            Global.exit(1);
+        };
+        defer ctx.allocator.free(package_json_contents);
+
+        const _current_version = extractVersion(package_json_contents);
```

Comment:
we don't need to worry about performance too much, it's fine to parse the json completely here. if it fails for any reason the current version can be excluded from the output text